src/Sparrow/Json/PerCoreContainer.cs

```diff
@@ -96,7 +122,23 @@ IEnumerator IEnumerable.GetEnumerator()
         public bool Remove(T item, (int, int) pos)
         {
             var array = _perCoreArrays[pos.Item1];
-            return Interlocked.CompareExchange(ref array[pos.Item2], null, item) == item;
+
+            if (Interlocked.CompareExchange(ref array[pos.Item2], null, item) == item)
+            {
+                Interlocked.Decrement(ref _perCoreArrayLength[pos.Item1].Value);
+                return true;
+            }
+
+            return false;
         }
+
+        
+    }
+
+    [StructLayout(LayoutKind.Explicit, Size = 64)]
+    internal struct PaddedInt
```

Comment:
nest it and mark as private?

Reply:
Generic types cannot have explicit layout.
When nest it inside, I'm getting the following error:
```
Unhandled exception. System.TypeLoadException: Could not load type 'PaddedInt' from assembly 'Sparrow, Version=6.0.107.0, Culture=neutral, PublicKeyToken=37f41c7f99471593' because generic types cannot have explicit layout.
   at Sparrow.Json.PerCoreContainer`1.TryPush(T cur)
   at Tryouts.Program.Main(String[] args) in C:\Work\ravendb-6.0\test\Tryouts\Program.cs:line 39
   at Tryouts.Program.<Main>(String[] args)
```